src/hooks/useValidation/useValidation.js

```diff
@@ -0,0 +1,28 @@
+import { useCallback } from 'react';
+import { validators } from './rules';
+
+const useValidation = (context) => {
+  const validate = useCallback((marcRecords) => {
+    const validationRules = validators[context.marcType][context.action];
+
+    const errorMessage = validationRules.reduce((_errorMessage, rule) => {
+      if (_errorMessage) {
+        return _errorMessage;
+      }
+
+      const error = rule.validator({ ...context, marcRecords }, rule);
+
+      if (error) {
+        return error;
+      }
+
+      return null;
+    }, null);
+
+    return errorMessage;
+  }, [context]);

```

Comment:
Context is always a new object and will trigger the useCallback continuously. Let's spread it. The rest looks cool.

Reply:
I think this hook doesn't need to know what is in the context - only tests need to know what properties there are if they need some additional data for validation.
If we add some property in our component, but forget to add it to dependencies here it could cause a bug, so I'll just remove `useCallback`

Reply:
I should have written more clearly, I meant `[...context]`. All context data will be cached as soon as all responses are retrieved (for example for the `linkingRules`).